src/appleseed/renderer/modeling/light/directionallight.cpp

```diff
@@ -242,7 +242,7 @@ namespace
 
             value = m_values.m_irradiance;
 
-            probability = 1.0f / (Pi<float>() * square(static_cast<float>(disk_radius)));
+            probability = 1.0f;
```

Comment:
That's only correct when `disk_radius` is 0.

You shouldn't change the way the probability is computed, it was correct before. What you need to do instead is no longer call `sample_disk()` when you have a target point. The code will also be much simpler in that case.